packages/@uppy/vue/src/Dashboard.vue

```diff
@@ -0,0 +1,62 @@
+<template>
+  <div ref="container" />
+</template>
+<script>
+import DashboardPlugin from '@uppy/dashboard'
+
+export default {
+  data () {
+    return {
+      plugin: null
+    }
+  },
+  props: {
+    uppy: {
+      required: true
+    },
+    props: {
+      type: Object
+    },
+    plugins: {
+      type: Array
+    }
+  },
+  mounted () {
+    this.installPlugin()
+  },
+  methods: {
+    installPlugin () {
+      const uppy = this.uppy
+      const options = {
+        id: 'vue:Dashboard',
+        inline: true,
+        plugins: this.plugins,
+        ...this.props,
+        target: this.$refs.container
+      }
+      uppy.use(DashboardPlugin, options)
+      this.plugin = uppy.getPlugin(options.id)
+    },
+    uninstallPlugin (uppy = this.uppy) {
+      uppy.removePlugin(this.plugin)
+    }
+  },
+  beforeDestroy () {
+    this.uninstallPlugin()
+  },
+  watch: {
+    uppy (current, old) {
+      if (old !== current) {
+        this.uninstallPlugin(old)
+        this.installPlugin()
+      }
+    },
+    props (current, old) {
+      if (JSON.stringify(old) !== JSON.stringify(current)) {
```

Comment:
This may not be quite right for options objects with functions in them. Something like https://www.npmjs.com/package/shallow-equal could work better.

Could we try using `this.plugin.setOptions(current)` here instead of re-installing the plugin?

Reply:
Yeah I'll go fix that

Reply:
Could I just get rid of the equality comparison altogether? I assume Uppy has some sort of internal mechanism to do it anyway

Reply:
The only possible "issue" is that Uppy rerenders when the options are changed, and this may not be necessary. But we could also address that inside the `setOptions()` implementation if it does turn out to be a problem (which it may not).

I think you need the commit hash for the preview. pressing the Y key on any source code page updates the URL to use the commit hash instead of branch name:
https://github.com/transloadit/uppy/blob/654a077f2cd388ef6c1497b9f4f7e2c69eea7124/packages/%40uppy/core/src/Plugin.js#L65-L68

Reply:
At least my thinking is, that `@uppy/react` also needs this comparison logic (actually, it still uses uninstall and install so that should probably be fixed at some point): https://github.com/transloadit/uppy/blob/2c264c3021adeb460aeb8cb6e968ec122b836d13/packages/%40uppy/react/src/Dashboard.js#L17-L22

So it *may* be worth putting that comparison logic in `@uppy/core`

Reply:
At least *for now* I setup `shallow-equal` in dashboard